Source/Dafny/DafnyOptions.cs

```diff
@@ -420,6 +423,16 @@ public enum IncludesModes { None, Immediate, Transitive }
         case "extractCounterexample":
           ExtractCounterexample = true;
           return true;
+
+        case "verificationLogger":
+          if (ps.ConfirmArgumentCount(1)) {
+            if (args[ps.i].Equals("trx")) {
+              VerificationLoggerConfig = args[ps.i];
+            } else {
+              throw new Exception("Invalid value for verificationLogger");
```

Comment:
I see that most of the other `throw new Exception...` statements in `DafnyOptions.cs` do it this same way, but it would be more useful to also output `args[ps.i]` in case of error. The reason is that the command-line parsing mechanisms allow both `/flag:arg` and `/flag arg`. If the user forgets to give `/verificationLogger` any argument at all, then it may not be immediately evident from the given command line what the invalid argument is. For example, `dafny /verificationLogger ProgFileA.dfy ProgFileB.dfy` would, unbeknownst to the uninitiated user, grab `ProgFileA.dfy` as the argument to `/verificationLogger`.

Also, I see that the other tests in this file use the format `args[ps.i].Equals("trx")`, but in C# this is equivalent to the easier-to-read `args[ps.i] == "trx"`.

Reply:
Not only that but throwing an exception leads to dumping a stack trace on the poor user for no good reason!

I've added a helper method to call `ps.Error` instead, which cleanly logs an error and moves on instead, and corrected the other places we throw exceptions to use it instead.

Reply:
I just noticed that I said "instead" no fewer than 3 times in one sentence, but rather than editing it to fix it I'm feeling perversely proud and will leave it for posterity. :)

Reply:
You mean "but _instead_ of editing it to fix it ..." :)